core/src/classes/codeConfig.ts

```diff
@@ -307,10 +311,14 @@ export function validateConfigObjects(
   for (const configObject of configObjects) {
     if (!configObject.id) {
       errors.push(
+        //@ts-ignore
         (configObject["name"]
-          ? `"${configObject["name"]}"`
-          : configObject["key"]
-          ? `"${configObject["key"]}"`
+          ? //@ts-ignore
+            `"${configObject["name"]}"`
+          : //@ts-ignore
+          configObject["key"]
+          ? //@ts-ignore
+            `"${configObject["key"]}"`
```

Comment:
This file has a lot of `@ts-ignore`... in a future PR we could come up with a bunch of typeGuards...

Reply:
Don't love this but unwrapping all these types would be a headache for sure.

Reply:
Yeah... the whole configLoader area is generally the weakest-typed part of the codebase because we don't know what type of configObject we are looking at until relatively far into the process